File: myapplication_ui/src/services/ApprovedByService.js
import http from "../http-common"; 

class ApprovedByService {
  getAllApprovedBys(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/approvedBy/approvedBys`, searchDTO);
  }

  get(approvedById) {
    return this.getRequest(`/approvedBy/${approvedById}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/approvedBy?field=${matchData}`, null);
  }

  addApprovedBy(data) {
    return http.post("/approvedBy/addApprovedBy", data);
  }

  update(data) {
  	return http.post("/approvedBy/updateApprovedBy", data);
  }
  
  uploadImage(data,approvedById) {
  	return http.postForm("/approvedBy/uploadImage/"+approvedById, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ApprovedByService();

File: src/main/java/com/myapplication/service/ApprovedByService.java
package com.myapplication.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.myapplication.domain.ApprovedBy;
import com.myapplication.dto.ApprovedByDTO;
import com.myapplication.dto.ApprovedBySearchDTO;
import com.myapplication.dto.ApprovedByPageDTO;
import com.myapplication.dto.ApprovedByConvertCriteriaDTO;
import com.myapplication.service.GenericService;
import com.myapplication.dto.common.RequestDTO;
import com.myapplication.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ApprovedByService extends GenericService<ApprovedBy, Integer> {

	List<ApprovedBy> findAll();

	ResultDTO addApprovedBy(ApprovedByDTO approvedByDTO, RequestDTO requestDTO);

	ResultDTO updateApprovedBy(ApprovedByDTO approvedByDTO, RequestDTO requestDTO);

    Page<ApprovedBy> getAllApprovedBys(Pageable pageable);

    Page<ApprovedBy> getAllApprovedBys(Specification<ApprovedBy> spec, Pageable pageable);

	ResponseEntity<ApprovedByPageDTO> getApprovedBys(ApprovedBySearchDTO approvedBySearchDTO);
	
	List<ApprovedByDTO> convertApprovedBysToApprovedByDTOs(List<ApprovedBy> approvedBys, ApprovedByConvertCriteriaDTO convertCriteria);

	ApprovedByDTO getApprovedByDTOById(Integer approvedById);







}





